File: src/main/kotlin/com/melowetty/hsepermhelper/models/LessonType.kt
package com.melowetty.hsepermhelper.models

enum class LessonType(val type: String, private val scheduleFilePattern: String) {
    LECTURE("Лекция", "{type}: {subject}") {
        override fun reformatSubject(subject: String): String {
            return subject
                .replace("(лекция)", "")
                .replace("(лекции)", "")
                .trim()
        }
    },
    SEMINAR("Семинар", "{type}: {subject}") {
        override fun reformatSubject(subject: String): String {
            return subject.replace("(семинар)", "").trim()
        }
    },
    EXAM("Экзамен", "{type}: {subject}") {
        override fun reformatSubject(subject: String): String {
            return subject.replace("ЭКЗАМЕН", "").trim()
        }
    },
    INDEPENDENT_EXAM("Независимый экзамен", "{subject}"),
    TEST("Зачёт", "{type}: {subject}") {
        override fun reformatSubject(subject: String): String {
            return subject.replace("ЗАЧЕТ", "").trim()
        }
    },
    PRACTICE("Практика", "{type}: {subject}") {
        override fun reformatSubject(subject: String): String {
            return if (subject == "ПРАКТИКА") ""
            else subject
        }
    },
    MINOR("Майнор", "{type}"),
    ENGLISH("Английский", "{type}"),
    STATEMENT("Ведомость", "{type}: {subject}"),
    ICC("МКД", "{type}: {subject}");
    fun toEventSubject(subject: String): String {
        return scheduleFilePattern
            .replace("{type}", type)
            .replace("{subject}", subject)
    }
    public open fun reformatSubject(subject: String): String {
        return subject
    }
}

File: model/lesson_type.py
import enum


class LessonType(enum.Enum):
    def __new__(cls, display_name: str, short_name: str):
        obj = object.__new__(cls)
        obj._value_ = display_name
        obj.display_name = display_name
        obj.short_name = short_name
        return obj

    LECTURE = ('лекция 😴', "лек.")
    SEMINAR = ('семинар 📗', "сем.")
    # SEMINAR = ('семинар 📓', "сем.") # halloween
    COMMON_MINOR = ('Майнор Ⓜ', "майнор")
    ENGLISH = ('английский 🆎', "англ.")
    EXAM = ('экзамен ☠️', "экз.")
    INDEPENDENT_EXAM = ('независимый экзамен ☠️☠️', "независим. экз.")
    TEST = ('зачёт ☠️', "зачёт")
    PRACTICE = ('практика 💼', "практ.")
    MINOR = ('Майнор Ⓜ', "майнор")
    COMMON_ENGLISH = ('английский 🆎', "англ.")
    STATEMENT = ('ведомость 📜', "ведомость")
    ICC = ('МКД 📙', "МКД")
    UNDEFINED_AED = ('ДОЦ по выбору 📕', "ДОЦ")
    AED = ('ДОЦ 📕', "ДОЦ")
    CONSULT = ('консультация 🗿', "консулт.")
    EVENT = ('мероприятие', "мероприятие")

    @classmethod
    def _missing_(cls, value):
        return None
